pkg/segment/reader/record/recordreader.go

```diff
@@ -233,42 +238,49 @@ func readAllRawRecords(orderedRecNums []uint16, blockIdx uint16, segReader *segr
 		mathColMap = make(map[string]int)
 	}
 
+	var isTsCol bool
 	for _, recNum := range orderedRecNums {
 		_, ok := results[recNum]
 		if !ok {
 			results[recNum] = make(map[string]interface{})
 		}
 
-		for _, colInfo := range segReader.AllColums {
-			col := colInfo.ColumnName
+		for colKeyIdx, cname := range allColKeyIndices {
 
-			_, ok := dictEncCols[col]
+			_, ok := dictEncCols[cname]
 			if ok {
 				continue
 			}
 
-			cValEnc, err := segReader.ExtractValueFromColumnFile(col, blockIdx, recNum, qid)
+			if cname == config.GetTimeStampKey() {
+				isTsCol = true
+			} else {
+				isTsCol = false
+			}
```

Comment:
nit: We don't need to the `else` block here. As by default `isTsCol` value is `false`.

Reply:
Same comment here also: https://github.com/siglens/siglens/pull/1428/files#diff-9ecb737019841f5110cc60e7192afc17333e9b7d86756fb829d607417d80683dR195-R197

Reply:
addressed in recent commit

Reply:
I'm pretty sure removing this `else` adds a bug. `isTsCol` is defined outside the loops, so once it gets set to `true` it's never reset to `false`. Am I missing something? You could also do
```
isTsCol = (cname == config.GetTimeStampKey())
```
unless you think that's less readable 

I'm just referring to the code here; I think the other code Mani linked is fine since there's not a loop